src/clj_excel/core.clj

```diff
@@ -326,4 +330,6 @@
   "Write workbook to output-stream as coerced by OutputStream."
   [^Workbook wb path]
   (with-open [out (output-stream path)]
-    (.write wb out)))
+    (.write wb out)
+    (when (isa? wb SXSSFWorkbook)
```

Comment:
Good catch. It probably should since dispose cleans up the temporary files created by the SXSSFWorkbook.